modules/common/datasource.go

```diff
@@ -4,19 +4,158 @@
 
 package common
 
+import (
+	"infini.sh/coco/core"
+	"infini.sh/framework/core/kv"
+	"infini.sh/framework/core/orm"
+	"infini.sh/framework/core/util"
+)
+
 type DataSource struct {
 	CombinedFullText
 
-	Type string `json:"type,omitempty" elastic_mapping:"type:{type:keyword}"` // Type of the datasource, eg: connector
-	Name string `json:"name,omitempty" elastic_mapping:"name:{type:keyword}"` // Display name of this datasource
-	Icon string `json:"icon,omitempty" elastic_mapping:"icon:{type:keyword}"` // Display name of this datasource
+	Type string `json:"type,omitempty" elastic_mapping:"type:{type:keyword,copy_to:combined_fulltext}"` // Type of the datasource, eg: connector
+	Name string `json:"name,omitempty" elastic_mapping:"name:{type:keyword,copy_to:combined_fulltext}"` // Display name of this datasource
+	Icon string `json:"icon,omitempty" elastic_mapping:"icon:{type:keyword}"`                           // Display name of this datasource
 
-	Connector ConnectorConfig `json:"connector,omitempty" elastic_mapping:"connector:{type:keyword}"`
+	Connector ConnectorConfig `json:"connector,omitempty" elastic_mapping:"connector:{type:object}"` // Connector configuration
 	// Whether synchronization is allowed
 	SyncEnabled bool `json:"sync_enabled" elastic_mapping:"sync_enabled:{type:keyword}"`
+	Enabled     bool `json:"enabled" elastic_mapping:"enabled:{type:keyword}"`
 }
 
 type ConnectorConfig struct {
 	ConnectorID string      `json:"id,omitempty" elastic_mapping:"id:{type:keyword}"`          // Connector ID for the datasource
 	Config      interface{} `json:"config,omitempty" elastic_mapping:"config:{enabled:false}"` // Configs for this Connector
 }
+
+const DisabledDatasourceIDsKey = "disabled_datasource_ids"
+
+// CacheDisabledDatasourceIDs retrieves all disabled data sources and caches their IDs.
+func CacheDisabledDatasourceIDs() error {
```

Comment:
I don't like the idea here to maintain such a complex logic to handle the id cache, why not just:
1. during build query condition, call a function: get_disabled_datasouces()
2. if it return any ids are not enabled, then append a extra must_not condion to the query builder

handle the cache login in get_disabled_datasouces only, if no cache, just call the backend api to fetch the result and cache it, for any datasource enable/disable api, just invalid the cache by the key, just as simple as that.